### PR TITLE
merging 

### PR DESCRIPTION
**Pull request by ahrenswett**

### CODE REVIEW

File: app/src/main/java/com/ahrenswett/classattendance/ClassAdapter.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        public void onClick(View view) {
+         listener.classCommand(this.myClass);
+        }
+    }
+
+    @NonNull
+    @Override
+    public ClassViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        Log.i(MainActivity.TAG, "IN onCreateViewHolder");
+        ConstraintLayout v = (ConstraintLayout) LayoutInflater.from(parent.getContext()).inflate(R.layout.fragment_class, parent,false);
```
Comment (arybin) [inline app/src/main/java/com/ahrenswett/classattendance/ClassAdapter.java:59]: why this cast to the `ConstraintLayout`?

File: app/src/main/java/com/ahrenswett/classattendance/ClassAdapter.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    public void onBindViewHolder(@NonNull ClassViewHolder holder, int position) {
+        Log.i(MainActivity.TAG, "IN onBindViewHolder");
+        Class classAtPosition = this.classes.get(position);
+        holder.ClassTitleView.setText(classAtPosition.getClassName());
+        holder.ClassSizeView.setText(classAtPosition.getSize());
+    }
+
+    @Override
+    public int getItemCount() {
+        return 0;
```
Comment (arybin) [inline app/src/main/java/com/ahrenswett/classattendance/ClassAdapter.java:73]: you should return the `classses.size()` here

File: app/src/main/java/com/ahrenswett/classattendance/ClassAdapter.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    public ClassViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        Log.i(MainActivity.TAG, "IN onCreateViewHolder");
+        ConstraintLayout v = (ConstraintLayout) LayoutInflater.from(parent.getContext()).inflate(R.layout.fragment_class, parent,false);
+        return new ClassViewHolder(v, listener);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ClassViewHolder holder, int position) {
+        Log.i(MainActivity.TAG, "IN onBindViewHolder");
+        Class classAtPosition = this.classes.get(position);
```
Comment (arybin) [inline app/src/main/java/com/ahrenswett/classattendance/ClassAdapter.java:66]: Suggestion: consider delegating setting text to your `ViewHolder` class. Create a method on your `ClassViewHolder` and just pass it the `Class`, let it do all the work

File: app/src/main/java/com/ahrenswett/classattendance/ClassAdapter.java (hunk 2 above)
Comment (ahrenswett) [reply app/src/main/java/com/ahrenswett/classattendance/ClassAdapter.java:73]: yeah that might help!

Comment (ahrenswett) [reply app/src/main/java/com/ahrenswett/classattendance/ClassAdapter.java:73]: that was a generated code and  forgot to change

File: app/src/main/java/com/ahrenswett/classattendance/ClassAdapter.java (hunk 3 above)
Comment (ahrenswett) [reply app/src/main/java/com/ahrenswett/classattendance/ClassAdapter.java:66]: so the lines just below this ended up being the problem had to add a 
holder.
`  @Override
    public void onBindViewHolder(@NonNull ClassViewHolder holder, int position) {
        Log.i(MainActivity.TAG, "IN onBindViewHolder");
        Class classAtPosition = this.classes.get(position);
        holder.myClass = classAtPosition;
        holder.ClassTitleView.setText(classAtPosition.getClassName());
//        holder.ClassSizeView.setText(classAtPosition.getSize());
    }`

couldnt tell until I changed getItem count as suggested above.

**Pull request merged**